orttraining/orttraining/test/python/orttraining_test_orttrainer_bert_toy_onnx.py

```diff
@@ -1,75 +1,515 @@
+            10.974218368530273, 10.970070838928223, 11.198983192443848, 11.134098052978516, 11.067017555236816]),
+    (1.0, optim.lr_scheduler.PolyWarmupLRScheduler, [0.18181818181818182, 0.06611570247933884, 0.03606311044327573, 0.026227716686018716, 0.02384337880547156,\
+            0.01607520271130791, 0.009693711967693117, 0.005062375970537139, 0.0021586043667598935, 0.0006508437050332076],
+            [10.988012313842773, 11.637386322021484, 11.099013328552246, 11.055734634399414, 11.145816802978516,\
+            10.974217414855957, 10.96664810180664, 11.193868637084961, 11.14560604095459, 11.097070693969727])
+])
+def testToyBERTModelLRScheduler(initial_lr, lr_scheduler, expected_learning_rates, expected_losses):
+    model_desc = bert_model_description()
+    model = load_bert_onnx_model()
+
+    total_steps = 10
+    optim_config = optim.LambConfig(lr=initial_lr)
+    opts =  orttrainer.ORTTrainerOptions({
+        'debug' : {
+            'deterministic_compute': True
+        },
+        'lr_scheduler' : lr_scheduler(total_steps=total_steps, warmup=0.5)
+    })
+    
+    torch.manual_seed(1)
+    set_seed(1)
+
+    trainer = orttrainer.ORTTrainer(model, model_desc, optim_config, options=opts)
+  
+    losses = []
+    learning_rates = []
+    for i in range(total_steps):
+        sample_input = generate_random_input_from_model_desc(model_desc, i)
+        losses.append(trainer.train_step(*sample_input).cpu().item())
+        learning_rates.append(trainer.options.lr_scheduler.get_last_lr()[0])
+    
+    _test_helpers.assert_model_outputs(learning_rates, expected_learning_rates)
+    _test_helpers.assert_model_outputs(losses, expected_losses, rtol=1e-6)
+    
+
+# TODO: Copy the expected values from legacy api
+# Dynamic Loss Scaler implemented implicitly
+@pytest.mark.parametrize("loss_scaler", [
+    (None),
+    (amp.DynamicLossScaler()),
+    (CustomLossScaler())
+])
+def testToyBERTModelMixedPrecisionLossScaler(loss_scaler):
+    total_steps = 10
+
+    model_desc = bert_model_description()
+    model = load_bert_onnx_model()
+
+    optim_config = optim.LambConfig()
+    opts =  orttrainer.ORTTrainerOptions({
+        'debug' : {
+            'deterministic_compute': True
+        },
+        'mixed_precision': {
+            'enabled': True,
+            'loss_scaler': loss_scaler
+        }
+    })
+    
+    trainer = orttrainer.ORTTrainer(model, model_desc, optim_config, options=opts)
+    
+    losses = []
+    for i in range(total_steps):
+        sample_input = generate_random_input_from_model_desc(model_desc, i)
+
+        losses.append(trainer.train_step(*sample_input).cpu().item())
+
+
+###############################################################################
+# Temporary tests comparing Legacy vs Experimental ORTTrainer APIs ############
+###############################################################################
+
+
+def testToyBERTModelLegacyExperimentalBasicTraining():
+    train_steps = 10
+
```

Comment:
```suggestion
device = 'cuda'
```
Define device here and *explicitly* use it for *both* APIs